docs/querying/api.md

```diff
@@ -363,6 +363,91 @@ $ curl http://localhost:9090/api/v1/targets
 }
 ```
 
+
+## Rules
+
+The `/rules` API endpoint returns a list of alerting and recording rules that
+are currently loaded. In addition it returns the currently active alerts fired
+by the Prometheus instance of each alerting rule.
+
+As the `/rules` endpoint is fairly new, it does not have the same stability
+guarantees as the overarching API v1.
+
+```
+GET /api/v1/rules
+```
+
+```json
+$ curl http://localhost:9090/api/v1/rules
+
+{
+    "data": {
+        "groups": [
+            {
+                "alertingRules": [
```

Comment:
There is an ordering across all rules, the alert recording and recording rules need to be in the same list.

Reply:
But giving up the `alertingRule` and recordingRule` bucket prevents us from differentiating the two, thereby we are not able to expose details like `Duration`, `Annotations`, active `Alerts`, ...

In which use case is the ordering of rules important?

Reply:
In any use case where there are interdependent rules and alerts.

Reply:
I think @brian-brazil has a point. I see several options.

1) add a `type` field that would be `alerting` or `recording`.

```json
{
    "data": {
        "groups": [
            {
                "rules": [
                    {
                        "alerts": [
                            {
                                "activeAt": "2018-07-04T20:27:12.60602144+02:00",
                                "annotations": {},
                                "labels": {
                                    "alertname": "my-alert"
                                },
                                "state": "firing",
                                "value": 1
                            }
                        ],
                        "annotations": {},
                        "duration": 0,
                        "labels": {},
                        "name": "my-alert",
                        "query": "vector(1)",
                        "type": "alerting"
                    }, {
                        "duration": 0,
                        "labels": {},
                        "name": "my-record",
                        "query": "vector(0)",
                        "type": "recording"
                    }
                ],
                "file": "",
                "interval": 60,
                "name": ""
            }
        ]
    },
    "status": "success"
}
```
2) Keep your proposal and add an `index` field.

```json
{
    "data": {
        "groups": [
            {
                "alertingRules": [
                    {
                        "alerts": [
                            {
                                "activeAt": "2018-07-04T20:27:12.60602144+02:00",
                                "annotations": {},
                                "labels": {
                                    "alertname": "my-alert"
                                },
                                "state": "firing",
                                "value": 1
                            }
                        ],
                        "annotations": {},
                        "duration": 0,
                        "index": 0,
                        "labels": {},
                        "name": "my-alert",
                        "query": "vector(1)"
                    }
                ],
                "file": "",
                "interval": 60,
                "name": "",
                "recordingRules": [{
                        "duration": 0,
                        "index": 1,
                        "labels": {},
                        "name": "my-record",
                        "query": "vector(0)",
                        "type": "recording"
                    }
                ]
            }
        ]
    },
    "status": "success"
}
```

I'd go with the first option as alerting rules are a superset of recording rules.